### DISCUSSION TITLE
Disable/enable swish for specific devices

### DISCUSSION THREAD

**Discussion by nianiam** [chrenn/swish-dl#7] [Ideas]
Using a trackpad without swish is almost impossible for me now.

However, when I am using my external monitors in clamshell mode I am using an external mouse. I use MacMouseFix for that, which (nowadays) emulates the trackpad in order to achieve it's functionality.

Rarely, but sometimes, I find myself accidentally scrolling over the top of a window, which activates Swish. This is almost always unintended, so it would be great to be able to disable swish for specific devices to prevent this from happening.

**Comment by chrenn** [comment]
That's tricky. I actually implemented mouse/trackpad detection similar to [ScrollReverser](https://github.com/pilotmoon/Scroll-Reverser). You can try it for yourself by setting `defaults write co.highlyopinionated.swish tapMagicMouse -bool false` via terminal.

However, the detection just stops working after a while. ScrollReverser gets around this by relaunching its process on wake, but that's not really feasible for Swish at the moment. I should probably submit a bug report to Apple, but CGEvent stuff rarely ever gets fixed.

**Reply by nianiam** [reply]
Hmm, that is annoying. How long is while? Is it triggered by a specific event (sleep etc)?

**Reply by chrenn** [reply]
Seems random to me, you'd have to try it for yourself.
